File: 2501-3000/2523.js
var closestPrimes = function(left, right) {
    let sieve = new Array(right + 1).fill(true);
    sieve[0] = sieve[1] = false;
    
    for (let i = 2; i * i <= right; i++) {
        if (sieve[i]) {
            for (let j = i * i; j <= right; j += i) {
                sieve[j] = false;
            }
        }
    }
    
    let primes = [];
    for (let i = left; i <= right; i++) {
        if (sieve[i]) {
            primes.push(i);
        }
    }
    
    if (primes.length < 2) {
        return [-1, -1];
    }
    
    let minGap = Infinity;
    let result = [-1, -1];
    
    for (let i = 1; i < primes.length; i++) {
        let gap = primes[i] - primes[i - 1];
        if (gap < minGap) {
            minGap = gap;
            result = [primes[i - 1], primes[i]];
        }
    }
    
    return result;
};

File: 2501-3000/2523.go
func closestPrimes(left int, right int) []int {
    sieve := make([]bool, right+1)
    for i := range sieve {
        sieve[i] = true
    }
    sieve[0], sieve[1] = false, false
    
    for i := 2; i*i <= right; i++ {
        if sieve[i] {
            for j := i * i; j <= right; j += i {
                sieve[j] = false
            }
        }
    }
    
    primes := []int{}
    for i := left; i <= right; i++ {
        if sieve[i] {
            primes = append(primes, i)
        }
    }
    
    if len(primes) < 2 {
        return []int{-1, -1}
    }
    
    minGap := int(^uint(0) >> 1)
    result := []int{-1, -1}
    
    for i := 1; i < len(primes); i++ {
        gap := primes[i] - primes[i-1]
        if gap < minGap {
            minGap = gap
            result[0] = primes[i-1]
            result[1] = primes[i]
        }
    }
    
    return result
}
